cpu/stm32l1/periph/i2c.c

```diff
@@ -467,99 +411,94 @@ int i2c_write_regs(i2c_t dev, uint8_t address, uint8_t reg, char *data, int leng
 void i2c_poweron(i2c_t dev)
 {
     switch (dev) {
-#if I2C_0_EN
+
         case I2C_0:
             I2C_0_CLKEN();
             break;
-#endif
-#if I2C_1_EN
+
         case I2C_1:
             I2C_1_CLKEN();
             break;
-#endif
     }
 }
 
 void i2c_poweroff(i2c_t dev)
 {
     switch (dev) {
-#if I2C_0_EN
+
         case I2C_0:
-            while (I2C_0_DEV->SR2 & I2C_SR2_BUSY);
+            while (i2c_config[dev].dev->SR2 & I2C_SR2_BUSY);
             I2C_0_CLKDIS();
             break;
-#endif
-#if I2C_1_EN
+
         case I2C_1:
-            while (I2C_0_DEV->SR2 & I2C_SR2_BUSY);
-            I2C_0_CLKDIS();
+            while (i2c_config[dev].dev->SR2 & I2C_SR2_BUSY);
+            I2C_1_CLKDIS();
             break;
-#endif
```

Comment:
If there was a way to automatically enable/disable the clock for the concerned device, one would save the whole switch case.